src/sdk/namespace_cache.js

```diff
@@ -739,6 +741,32 @@ class NamespaceCache {
         return this.namespace_hub.get_blob_block_lists(params, object_sdk);
     }
 
+    //////////
+    // ACLs //
+    //////////
+
+    async get_object_acl(params, object_sdk) {
+        if (config.NAMESPACE_CACHING.ACL_HANDLING === "reject") {
+            throw new S3Error(S3Error.AccessDenied);
+        }
+
+        if (config.NAMESPACE_CACHING.ACL_HANDLING === "pass-through") {
+            return this.namespace_hub.get_object_acl(params, object_sdk);
+        }
+
+        await this.read_object_md(params, object_sdk);
+        return s3_utils.DEFAULT_OBJECT_ACL;
+    }
+
+    async put_object_acl(params, object_sdk) {
+        if (config.NAMESPACE_CACHING.ACL_HANDLING === "reject") {
+            throw new S3Error(S3Error.AccessDenied);
+        }
+
+        if (config.NAMESPACE_CACHING.ACL_HANDLING === "pass-through") {
+            return this.namespace_hub.put_object_acl(params, object_sdk);
+        }
+    }
```

Comment:
put_object_acl when ACL_HANDLING === '' then we also need to load the object for the errors:
```
await this.read_object_md(params, object_sdk);
```